content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -0,0 +1,1260 @@
+allow you to change a query into different query types as long as the
+componenets accessed are a subset of the original query. If you do try to access
+data that is not in the original query, this method will panic.
+
+```rust
+fn reusable_function(lens: &mut QueryLens<&Transform>) {
+    let query = lens.query();
+    // do something with the query...
+}
+
+// We can use the function in a system that takes the exact query.
+fn system_1(mut query: Query<&Transform>) {
+    reusable_function(&mut query.as_query_lens());
+}
+
+// We can also use it with a query that does not match exactly
+// by transmuting it.
+fn system_2(mut query: Query<(&mut Transform, &Velocity), With<Enemy>>) {
+    let mut lens = query.transmute_lens::<&Transform>();
+    reusable_function(&mut lens);
+}
+```
+
+Note that the `QueryLens` will still iterate over the same entities as the
+original `Query` it is derived from. A `QueryLens<&Transform>` taken from
+a `Query<(&Transform, &Velocity)>`, will only include the `Transform` of
+entities with both `Transform` and `Velocity` components.
+
+Besides removing parameters you can also change them in limited ways to the
+different smart pointer types. One of the more useful is to change a
+`& mut` to a `&`. See the [documentation](https://docs.rs/bevy/latest/bevy/ecs/system/struct.Query.html#method.transmute_lens)
+for more details.
+
+One thing to take into consideration is the transmutation is not free.
+It works by creating a new state and copying a bunch of the cached data
+inside the original query. It's not a expensive operation, but you should
+probably avoid doing it inside a hot loop.
+
+## Entity Optimizations
+
+<div class="release-feature-authors">authors: @Bluefinger, @notverymoe, @scottmcm, @bushrat011899, @james7132</div>
+
+`Entity` (Bevy's 64-bit unique identifier for enitities) received a number of changes this cycle, combining laying some more groundwork for relations alongside _related_, and nice to have, performance optimizations. The work here involved a lot of deep-diving into compiler codegen/assembly output, with running lots of benchmarks and testing in order to ensure all changes didn't cause breakages or major problems. Although the work here was dealing with mostly _safe_ code, there were lots of underlying assumptions being changed that could have impacted code elsewhere. This was the most "micro-optimization" oriented set of changes in Bevy 0.13.
+
+### Changing `Entity`'s Layout
+
+This is a story told in two parts, with part one dealing with two PRs: The "Unified Identifier for entities & relations" PR ([#9797](https://github.com/bevyengine/bevy/pull/9797) by @Bluefinger) and "Change Entity::generation from u32 to NonZeroU32 for niche optimization" ([#9907](https://github.com/bevyengine/bevy/pull/9907) by @notverymoe). Fundamentally, they both involved changing the layout and assumptions of `Entity`, unlocking a building blocks for relations and improving memory/codegen for `Entity`.
```

Comment:
```suggestion
This is a story told in two parts, with part one dealing with two PRs: The "Unified Identifier for entities & relations" PR ([#9797](https://github.com/bevyengine/bevy/pull/9797) by @Bluefinger) and "Change Entity::generation from u32 to NonZeroU32 for niche optimization" ([#9907](https://github.com/bevyengine/bevy/pull/9907) by @notverymoe). Fundamentally, they both involved changing the layout and assumptions of `Entity`, unlocking building blocks for relations and improving memory/codegen for `Entity`.
```
Singular/Plural mix here, probably better to remove the "a". Maybe even use "some" instead?